lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -500,6 +501,39 @@ def test__has_access_course_can_enroll(self):
         )
         assert not access._has_access_course(user, 'enroll', course)
 
+    @override_settings(COURSES_INVITE_ONLY=False)
+    def test__course_default_invite_only_flag_false(self):
+        """Tests that default value of COURSES_INVITE_ONLY as False."""
+
+        user = UserFactory.create()
+
+        course = self._mock_course(invitation=True)
+        self.assertFalse(access._has_access_course(user, 'enroll', course))
+
+        course = self._mock_course(invitation=False)
+        self.assertTrue(access._has_access_course(user, 'enroll', course))
+
+    @override_settings(COURSES_INVITE_ONLY=True)
+    def test__course_default_invite_only_flag_true(self):
+        """Tests that default value of COURSES_INVITE_ONLY as True."""
+
+        user = UserFactory.create()
+
+        course = self._mock_course(invitation=True)
+        self.assertFalse(access._has_access_course(user, 'enroll', course))
+
+        course = self._mock_course(invitation=False)
+        self.assertFalse(access._has_access_course(user, 'enroll', course))
+
+    def _mock_course(self, invitation):
```

Comment:
Since this is specific to your tests, I wonder if it should have a more specific name such as `_mock_course_with_invitation` or something better than that, ha